extensions/amp-share-tracking/0.1/test/test-amp-share-tracking.js

```diff
@@ -14,213 +14,205 @@
  * limitations under the License.
  */
 
-import {createIframePromise} from '../../../../testing/iframe';
 import {AmpShareTracking} from '../amp-share-tracking';
-import {Viewer} from '../../../../src/service/viewer-impl';
+import {History} from '../../../../src/service/history-impl';
 import {Xhr} from '../../../../src/service/xhr-impl';
 import {shareTrackingForOrNull} from '../../../../src/share-tracking-service';
 import {toggleExperiment} from '../../../../src/experiments';
-import * as sinon from 'sinon';
 import * as bytes from '../../../../src/utils/bytes';
 
-describe('amp-share-tracking', () => {
-  let sandbox;
-  let viewerGetFragmentStub;
-  let viewerUpdateFragmentStub;
+describes.fakeWin('amp-share-tracking', {
+  amp: {
+    ampdoc: 'single',
+    extensions: ['amp-share-tracking'],
+  },
+}, env => {
+  let win;
+  let ampdoc;
+  let historyGetFragmentStub;
+  let historyUpdateFragmentStub;
   let xhrStub;
   let randomBytesStub;
 
   beforeEach(() => {
-    sandbox = sinon.sandbox.create();
-    viewerGetFragmentStub = sandbox.stub(Viewer.prototype, 'getFragment');
-    viewerUpdateFragmentStub = sandbox.stub(Viewer.prototype, 'updateFragment');
+    win = env.win;
+    ampdoc = env.ampdoc;
+    historyGetFragmentStub = sandbox.stub(History.prototype, 'getFragment');
+    historyUpdateFragmentStub = sandbox.stub(History.prototype,
+        'updateFragment');
     xhrStub = sandbox.stub(Xhr.prototype, 'fetchJson');
     randomBytesStub = sandbox.stub(bytes, 'getCryptoRandomBytesArray');
+    toggleExperiment(win, 'amp-share-tracking', true);
   });
 
   afterEach(() => {
-    sandbox.restore();
+    toggleExperiment(win, 'amp-share-tracking', false);
   });
 
   function getAmpShareTracking(optVendorUrl) {
-    return createIframePromise(/* runtimeOff */ true).then(iframe => {
-      toggleExperiment(iframe.win, 'amp-share-tracking', true);
-      const el = iframe.doc.createElement('amp-share-tracking');
-      if (optVendorUrl) {
-        el.setAttribute('data-href', optVendorUrl);
-      }
-      iframe.doc.body.appendChild(el);
-      const ampShareTracking = new AmpShareTracking(el);
-      ampShareTracking.buildCallback();
-      return ampShareTracking;
-    });
+    const element = win.document.createElement('amp-share-tracking');
```

Comment:
If you call `getAmpShareTracking` for all tests - it'd be better to just merge this code into `beforeEach` and expose `ampShareTracking` as a `describe`'s let-variable.